### PR TITLE
Rally benchmark kubernetes.container

### PR DESCRIPTION
**Pull request by aspacca**
Enhancement
## Proposed commit message
Add artifacts for `elastic-package` rally benchmark

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
~- [ ] I have added an entry to my package's `changelog.yml` file.~
~- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).~

## Author's Checklist

- [ ] 

## How to test this PR locally
From `kubernets` package root (remember to bring up the elastic-package stack before):
`./elastic-package benchmark rally --benchmark container-benchmark -v`

## Related issues

- 

## Screenshots
```
--- Benchmark results for package: kubernetes - START ---
╭─────────────────────────────────────────────────────────────────────────────────────────────────╮
│ info                                                                                            │
├────────────────────────┬────────────────────────────────────────────────────────────────────────┤
│ benchmark              │                                                    container-benchmark │
│ description            │                   Benchmark 20000 kubernetes.container events ingested │
│ run ID                 │                                   2c29c961-f833-48b2-9175-821d60152507 │
│ package                │                                                             kubernetes │
│ start ts (s)           │                                                             1699527229 │
│ end ts (s)             │                                                             1699527262 │
│ duration               │                                                                    33s │
│ generated corpora file │ /Users/andreaspacca/.elastic-package/tmp/rally_corpus/corpus-144197238 │
╰────────────────────────┴────────────────────────────────────────────────────────────────────────╯
╭─────────────────────────────────────────────────────────────────────────╮
│ parameters                                                              │
├─────────────────────────────────┬───────────────────────────────────────┤
│ package version                 │                                1.53.0 │
│ data_stream.name                │                             container │
│ corpora.generator.total_events  │                                 20000 │
│ corpora.generator.template.path │ ./container-benchmark/template.ndjson │
│ corpora.generator.template.raw  │                                       │
│ corpora.generator.template.type │                                gotext │
│ corpora.generator.config.path   │      ./container-benchmark/config.yml │
│ corpora.generator.config.raw    │                                 map[] │
│ corpora.generator.fields.path   │      ./container-benchmark/fields.yml │
│ corpora.generator.fields.raw    │                                 map[] │
╰─────────────────────────────────┴───────────────────────────────────────╯
╭───────────────────────╮
│ cluster info          │
├───────┬───────────────┤
│ name  │ elasticsearch │
│ nodes │             1 │
╰───────┴───────────────╯
╭──────────────────────────────────────────────────────────────╮
│ data stream stats                                            │
├────────────────────────────┬─────────────────────────────────┤
│ data stream                │ metrics-kubernetes.container-ep │
│ approx total docs ingested │                           20000 │
│ backing indices            │                               2 │
│ store size bytes           │                        20367390 │
│ maximum ts (ms)            │                   1699530829168 │
╰────────────────────────────┴─────────────────────────────────╯
╭────────────────────────────────────╮
│ disk usage for index .ds-metrics-k │
│ ubernetes.container-ep-2023.11.09- │
│ 000001 (for all fields)            │
├──────────────────────────────┬─────┤
│ total                        │ 0 B │
│ inverted_index.total         │ 0 B │
│ inverted_index.stored_fields │ 0 B │
│ inverted_index.doc_values    │ 0 B │
│ inverted_index.points        │ 0 B │
│ inverted_index.norms         │ 0 B │
│ inverted_index.term_vectors  │ 0 B │
│ inverted_index.knn_vectors   │ 0 B │
╰──────────────────────────────┴─────╯
╭───────────────────────────────────────╮
│ disk usage for index .ds-metrics-kube │
│ rnetes.container-ep-2023.11.09-000002 │
│ (for all fields)                      │
├──────────────────────────────┬────────┤
│ total                        │  20 MB │
│ inverted_index.total         │ 5.0 MB │
│ inverted_index.stored_fields │ 7.5 MB │
│ inverted_index.doc_values    │ 6.7 MB │
│ inverted_index.points        │ 337 kB │
│ inverted_index.norms         │    0 B │
│ inverted_index.term_vectors  │    0 B │
│ inverted_index.knn_vectors   │    0 B │
╰──────────────────────────────┴────────╯
╭───────────────────────────────────────────────────────────────────────────────────────╮
│ pipeline metrics-kubernetes.container-1.53.0 stats in node -u0THNwnRLeH1Qslb_aclw     │
├────────────────────────────────────────────────┬──────────────────────────────────────┤
│ Totals                                         │ Count: 20000 | Failed: 0 | Time: 9ms │
│ pipeline (metrics-kubernetes.container@custom) │ Count: 20000 | Failed: 0 | Time: 3ms │
╰────────────────────────────────────────────────┴──────────────────────────────────────╯
╭─────────────────────────────────────────────────────────────────────────────────────────────╮
│ rally stats                                                                                 │
├────────────────────────────────────────────────────────────────┬────────────────────────────┤
│ Cumulative indexing time of primary shards                     │     0.3373833333333333 min │
│ Min cumulative indexing time across primary shards             │                      0 min │
│ Median cumulative indexing time across primary shards          │  0.0057666666666666665 min │
│ Max cumulative indexing time across primary shards             │    0.06888333333333334 min │
│ Cumulative indexing throttle time of primary shards            │                      0 min │
│ Min cumulative indexing throttle time across primary shards    │                      0 min │
│ Median cumulative indexing throttle time across primary shards │                      0 min │
│ Max cumulative indexing throttle time across primary shards    │                      0 min │
│ Cumulative merge time of primary shards                        │                0.02105 min │
│ Cumulative merge count of primary shards                       │                         48 │
│ Min cumulative merge time across primary shards                │                      0 min │
│ Median cumulative merge time across primary shards             │                0.00075 min │
│ Max cumulative merge time across primary shards                │  0.0021833333333333336 min │
│ Cumulative merge throttle time of primary shards               │                      0 min │
│ Min cumulative merge throttle time across primary shards       │                      0 min │
│ Median cumulative merge throttle time across primary shards    │                      0 min │
│ Max cumulative merge throttle time across primary shards       │                      0 min │
│ Cumulative refresh time of primary shards                      │                 0.0553 min │
│ Cumulative refresh count of primary shards                     │                       1572 │
│ Min cumulative refresh time across primary shards              │                      0 min │
│ Median cumulative refresh time across primary shards           │  0.0011833333333333333 min │
│ Max cumulative refresh time across primary shards              │   0.015716666666666667 min │
│ Cumulative flush time of primary shards                        │     0.8749166666666667 min │
│ Cumulative flush count of primary shards                       │                       1414 │
│ Min cumulative flush time across primary shards                │ 3.3333333333333335e-05 min │
│ Median cumulative flush time across primary shards             │   0.028866666666666665 min │
│ Max cumulative flush time across primary shards                │    0.07048333333333333 min │
│ Total Young Gen GC time                                        │                    0.037 s │
│ Total Young Gen GC count                                       │                          8 │
│ Total Old Gen GC time                                          │                        0 s │
│ Total Old Gen GC count                                         │                          0 │
│ Store size                                                     │      0.0381554514169693 GB │
│ Translog size                                                  │    0.018298085778951645 GB │
│ Heap used for segments                                         │                       0 MB │
│ Heap used for doc values                                       │                       0 MB │
│ Heap used for terms                                            │                       0 MB │
│ Heap used for norms                                            │                       0 MB │
│ Heap used for points                                           │                       0 MB │
│ Heap used for stored fields                                    │                       0 MB │
│ Segment count                                                  │                        343 │
│ Total Ingest Pipeline count                                    │                      20032 │
│ Total Ingest Pipeline time                                     │                    0.893 s │
│ Total Ingest Pipeline failed                                   │                          0 │
│ Min Throughput                                                 │             1924.00 docs/s │
│ Mean Throughput                                                │             1924.00 docs/s │
│ Median Throughput                                              │             1924.00 docs/s │
│ Max Throughput                                                 │             1924.00 docs/s │
│ 50th percentile latency                                        │       1091.372187500001 ms │
│ 100th percentile latency                                       │              1639.98475 ms │
│ 50th percentile service time                                   │       1091.372187500001 ms │
│ 100th percentile service time                                  │              1639.98475 ms │
│ error rate                                                     │                     0.00 % │
╰────────────────────────────────────────────────────────────────┴────────────────────────────╯

--- Benchmark results for package: kubernetes - END   ---
Done

```

### CODE REVIEW

**Comment by tommyers-elastic** [comment]
LGTM - do the fields, templates etc need to be kept in sync with the rest of the package? i.e. if we add a new field, or modify a mapping, should we update the corresponding files here?

**Comment by aspacca** [comment]
> LGTM - do the fields, templates etc need to be kept in sync with the rest of the package? i.e. if we add a new field, or modify a mapping, should we update the corresponding files here?

indeed yes

**Comment by gizas** [comment]
@aspacca I added some more changes after discussing with @martijnvg 
See here: [Setting same pod.name and pod.uids values](https://github.com/elastic/elastic-integration-corpus-generator-tool/pull/111/commits/46f451f96be3e1c7c166a12bf099b04fa11dfd9d)

**Comment by aspacca** [comment]
> @aspacca I added some more changes after discussing with @martijnvg See here: [Setting same pod.name and pod.uids values](https://github.com/elastic/elastic-integration-corpus-generator-tool/pull/111/commits/46f451f96be3e1c7c166a12bf099b04fa11dfd9d)

@gizas how many events do you generate? what's the timestamp difference between the first and the last events? two days, correct?

**Comment by gizas** [comment]
> > @aspacca I added some more changes after discussing with @martijnvg See here: [Setting same pod.name and pod.uids values](https://github.com/elastic/elastic-integration-corpus-generator-tool/pull/111/commits/46f451f96be3e1c7c166a12bf099b04fa11dfd9d)
> 
> @gizas how many events do you generate? what's the timestamp difference between the first and the last events? two days, correct?

Events are 8640000
And yes two days (See https://github.com/elastic/elastic-integration-corpus-generator-tool/pull/111/files#diff-d34fb1cf4866915a96be5b7ef33896bfecb1d98d1571dc32204396cc8aef5255L45)

**Comment by aspacca** [comment]
@gizas all good here? :)

I've applied the latest change related to node uid

File: packages/kubernetes/_dev/benchmark/rally/container-benchmark/template.ndjson [hunk 1]
```diff
@@ -0,0 +1,227 @@
+{{- $period := generate "metricset.period" }}
+{{- $agentId := generate "agent.id" }}
+{{- $agentVersion := generate "agent.version" }}
+{{- $agentName := generate "agent.name" }}
+{{- $agentEphemeralid := generate "agent.ephemeral_id" }}
+{{- $container_uId := uuidv4 }}
```
Comment (gizas) [inline packages/kubernetes/_dev/benchmark/rally/container-benchmark/template.ndjson:6]: This is not needed if we remove UIDs from the whole document

File: packages/kubernetes/_dev/benchmark/rally/container-benchmark/template.ndjson [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+   "container":{
+      "memory":{
+         "usage": {{divf $pct 1000000}}
+      },
+      "name":"{{ $name }}",
+      "runtime":"containerd",
+      "cpu":{
+         "usage": {{divf $pct 1000000}}
+      },
+      "id":"{{ $container_uId }}"
```
Comment (gizas) [inline packages/kubernetes/_dev/benchmark/rally/container-benchmark/template.ndjson:23]: Please lets remove the actual uid here and follow: https://github.com/elastic/elastic-integration-corpus-generator-tool/blob/9ef0f5c6367522178b7081b8fa90fe84bcc023a8/assets/templates/kubernetes.container/schema-b/gotext_multiline.tpl#L26

**Pull request merged**